### ISSUE TITLE
C++ iterators for Lua tables

### ISSUE THREAD

**Issue by thinlizzy** [Rapptz/sol#72]
Any plans on adding iteration capabilities in sol::table ?

**Comment by ThePhD** [comment]
Try using

``` cpp
sol::table t = ...;
t.for_each( /* function that takes key, value */ ) ;
```

Note that this is only present in the `develop` branch of `sol`.

**Comment by thinlizzy** [comment]
error: 'class sol::table' has no member named 'for_each'

On Sun, Feb 14, 2016 at 7:57 PM, The Phantom Derpstorm <
notifications@github.com> wrote:

> Try using
> 
> sol::table t = ...;
> t.for_each( /\* function that takes key, value */ ) ;
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Rapptz/sol/issues/72#issuecomment-184015985.

## 

https://github.com/thinlizzy/ http://die-xml.googlecode.com/ - C++11 and
C++14 open source wrappers and libraries for various purposes (XML, image,
webserver... and even a MTG implementation)
http://baudejogos.net - o único site brasileiro decente de jogos

**Comment by ThePhD** [comment]
Please pull from the _develop_ branch of `Rapptz/sol` (you're currently using the `master` branch) to get more updated features, or use the experimental version 2 of the library [here](https://github.com/ThePhD/sol2).

**Comment by thinlizzy** [comment]
Hi! Does the develop branch also support checking if a table key exists?

I've seen sol2 has sol::optional, which also works for me. Perhaps I should early migrate to sol2, instead of trying the develop branch?

**Comment by ThePhD** [comment]
Yes. sol2 is no longer experimental and has support for the features  you're looking for.
